File: python/codeforces/ClubsMela1.py

n, q = map(int, input().split())
maxi = [-2] *(n+1)

for i in range(n):
    a, b = map(int, input().split())
    maxi[b] = max(b, maxi[a])

for i in range(q):
    x = int(input())
    print(maxi[x])

File: python/codeforces/ClubsMela1.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int n, q;
    cin>>n>>q;

    vector<int> maxi(n+1, -1);
    for(int i=0; i<n+1; i++) {
		maxi[i] = -1;
    }
    
    for(int i=0; i<n; i++) {
    	int a, b;
    	cin>>a>>b;
    	maxi[b] = max(maxi[a], b);
    }
    
    for(int i=0; i<n; i++) {
    	int x;
    	cin>>x;
    	cout<<maxi[x]<<endl;
    }
}